### ISSUE TITLE
Adding Ableton Link synchronisation to TempoClock

### ISSUE THREAD

**Issue by smiarx** [supercollider/supercollider#3347]
I needed support for [Ableton Link](https://github.com/Ableton/link) in Supercollider for a live project.  [LinkUGen](https://github.com/byulparan/LinkUGen) already exists but it only works on the server side, and I usually work with Patterns and Routines. Unfortunately quarks don't allow hard-coded C++ extensions of sclang, so I created a [fork](https://github.com/smiarx/supercollider/tree/ableton-link-3.9) implementing it.

It adds some primitives and a method `TempoClock.link(state=true)` that allow any TempoClock beats and tempo to be synchronized to the current Link sessions. One way to test it is to create a pattern with quant, removing server latency.
```
t = TempoClock(90/60).link;
Pbind(\degree, Pseq([0,1,2,3],inf), \latency, 0).play(t, quant:4);
```
This should be synchronized with every other Link application.

It also add the variables `TempoClock.numPeers` which is set to the number of connected peers in the session, and `TempoClock.linkTempoChanged` which can be set to a callback function called when the tempo is changed in the session by other users.

Since adding this as a quark is impossible right now, I'd like to know the team opinion on this and if it's worth making a pull request. Many people work with Live and more and more application support Link, I think it would be a great addition to SuperCollider.

**Comment by snappizz** [comment]
wonderful!! i think this would be a perfectly appropriate feature to introduce to core for SC 3.10.

**Comment by telephon** [comment]
very nice, yes, please make a pull request! Then we can discuss the details about what the best code interface is.

**Comment by brianlheim** [comment]
Cool! I also support moving this to core.

**Comment by smiarx** [comment]
Pull request is here #3351

**Comment by brianlheim** [comment]
Closed by #4331 , thanks again @smiarx for getting this started!
